[6기 봉원희] Mission 4 댓글 기능 구현

2021bong opened the pull request:
- 댓글 입력칸에 이벤트 등록
- 게시 버튼 클릭 이벤트 등록
- 댓글 입력 함수 작성

미션 4까지 기능 구현했습니다! 😀

Commit:
Add : Mission 4 댓글 기능 구현
- 댓글 입력칸에 이벤트 등록
- 게시 버튼 클릭 이벤트 등록
- 댓글 입력 함수 작성

Commit:
Remove : kimcode 폴더 삭제

markkim340 commented on wonnybong/main.html:
```diff
@@ -0,0 +1,363 @@
+          </li>
+        </ul>
+      </nav>
+    </header>
+    <!-- //header -->
+    <!-- main -->
+    <main class="main-setting">
```
주석으로 구분되어 가독성이 좋습니다.

markkim340 commented on wonnybong/js/main.js:
```diff
@@ -0,0 +1,42 @@
+const user = "2021Bong";
+const commentTimeArr = [...document.getElementsByClassName("post-act-time")];
+
+const getComment = (e) => {
+  let inputText = e.target.value;
+  let btn = e.target.form[1];
+  if (inputText) {
+    btn.disabled = false;
+    btn.classList.add("abled");
+  } else {
+    btn.disabled = true;
+    btn.classList.remove("abled");
+  }
+};
+
```
들여쓰기가 잘 되어있어 이해하기 좋습니다.

markkim340 commented on wonnybong/style/login.css:
```diff
@@ -0,0 +1,77 @@
+  font-size: 45px;
+}
+
+,
+.login-form {
+  margin-top: 20px;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+}
+
```
체크리스트에 있는 css 속성 순서에 맞추어 작성하면 좋을 것 같습니다 ^^

markkim340 reviewed (commented):
수고하셨습니다
벌써 미션4까지 완성도 있게 잘 수행하셨네요..!
남은기간도 화이팅!

gyoeun666 commented on wonnybong/main.html:
```diff
@@ -0,0 +1,363 @@
+              />
+            </a>
+          </li>
+          <li>
+            <a href="#">
+              <img
+                class="icon-setting"
+                alt="게시글 작성"
+                src="./img/plus_icon.png"
+              />
+            </a>
+          </li>
+          <li>
+            <a href="#">
+              <img
+                class="icon-setting"
+                alt="탐색"
+                src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/explore.png"
+              />
+            </a>
+          </li>
+          <li>
+            <a href="#">
+              <img
+                class="icon-setting"
+                alt="소식"
+                src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/heart.png"
+              />
+            </a>
+          </li>
+          <li>
+            <a href="#">
+              <img
+                class="icon-setting"
+                alt="프로필"
+                src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/profile.png"
+              />
+            </a>
+          </li>
+        </ul>
```
img 태그로 인해 코드가 너무 길어지는 감이 있어 css로 넣는것도 괜찮을 것 같습니다 😄

gyoeun666 commented on wonnybong/main.html:
```diff
@@ -0,0 +1,363 @@
+            <span class="bold mr5">2021Bong</span>공부 화이팅~<a
+              href="#"
+              class="feed-cont-more ml5"
+              >...더 보기</a
```
text-overflow: ellipsis 이용해서 말줄임 표현할 수 있다고 하네요! 한번 찾아보시면 좋을 것 같습니다😄

2021bong replied:
text-overflow: ellipsis를 사용할 때  white-space: nowrap 속성을 같이 사용해줘야 ...이 나타나더라고요! 참고하시면 좋을 것 같습니다!

gyoeun666 commented on wonnybong/main.html:
```diff
@@ -0,0 +1,363 @@
+            </ul>
+          </div>
+        </div>
+        <!-- feed content container -->
+        <div class="feed-cont">
+          <p class="bold">좋아요 56,782개</p>
+          <p>
+            <span class="bold mr5">2021Bong</span>공부 화이팅~<a
+              href="#"
+              class="feed-cont-more ml5"
+              >...더 보기</a
+            >
+          </p>
+          <a href="#" class="feed-cont-more">댓글 24개 모두 보기</a>
+          <p class="feed-cont-time">1시간 전</p>
+        </div>
+        <!-- feed comment container -->
```
주석으로 구분해두셔서 한눈에 어떤 영역인지 보이네요!👍

gyoeun666 reviewed (commented):
고생하셨습니다! 😄

Commit:
Refactor : 코드 리뷰 반영
- css 속성 순서 정리
- .feed-cont-com에 text-overflow: ellipsis 속성 추가

royggle commented on wonnybong/js/main.js:
```diff
@@ -0,0 +1,42 @@
+  // console.log(newComment);
+  const commentTime = commentTimeArr[i];
+  const commentContainer = commentTime.parentNode;
+  commentContainer.insertBefore(newComment, commentTime);
```
코드 정말 보기 좋네요ㅎㅎ 미리 정답 본 느낌.. 많이 배웠습니다!

royggle commented on wonnybong/style/reset.css:
```diff
@@ -0,0 +1,376 @@
+/*! normalize.css v8.0.1 | MIT License | github.com/necolas/normalize.css */
+
```
리셋까지 준비하시다니.. 고생하셨습니다!

Commit:
Add : Mission 5 완료
- feeds.json 파일 생성
- fetch 후 데이터로 html 내용 동적으로 변경

Commit:
Add : Mission 5 리소스 추가
- board2_img 추가
- board3_img 추가

Commit:
Refactor: common.css 추가
- 정렬관련 css 클래스 common.css로 분리

daydreamplace commented on wonnybong/main.html:
```diff
@@ -0,0 +1,357 @@
+    </header>
+    <!-- //header -->
+    <!-- main -->
+    <main class="main-setting">
```
html 만들면서 구분 불편했는데 이런 방법도 있군여 하나 배워갑니다..!

Commit:
Add : Mission 6 ID, PW 유효성 검증
- id '@' 포함, pw 5글자 이상 유효성 검증 추가
- 기본 quote style single quote로 변경

Commit:
Add : Mission 7 댓글 좋아요, 삭제 기능 추가
- 댓글 생성시 하트 추가 및 좋아요 기능 구현
- 댓글 생성시 삭제 아이콘 추가 및 삭제 기능 구현
- 댓글 삭제 전 comfirm 단계 구현

daydreamplace commented on wonnybong/style/main.css:
```diff
@@ -0,0 +1,257 @@
+  height: 32px;
+  border-radius: 50%;
+  background: url('../img/profile.png') no-repeat 15%;
+  margin-right: 10px;
```
css 권장순서로 변경하시면 좋을 것 같아요!

Commit:
Add : Mission 8 아이디 검색 기능
- profiles.json 파일 생성
- input focus시 검색 토글 생기도록 구현
- fetch 후 검색 토글 리스트 생성
- input focus시에만 보이는 delete 버튼 구현

Commit:
Add : Mission 9 nav의 프로필 클릭시 메뉴 박스 생성
- nav의 프로필 클릭시 메뉴 토글 구현
- 로그아웃 버튼 클릭시 로그인 페이지로 이동

fullkeem commented on wonnybong/js/login.js:
```diff
@@ -0,0 +1,22 @@
+function ableLoginBtn() {
+  let idValue = uidInput.value;
+  let pwValue = upwInput.value;
+  if (idValue.includes('@') && pwValue.length >= 5) {
+    loginBtn.disabled = false;
+    loginBtn.classList.add('abled');
+  } else {
+    loginBtn.disabled = true;
+    loginBtn.classList.remove('abled');
```
classList 몰랐는데 덕분에 알게 되었습니다!
감사합니다ㅎ

fullkeem commented on wonnybong/style/main.css:
```diff
@@ -0,0 +1,358 @@
+  font-family: 'Lobster', cursive;
+}
+
+.search {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  width: 268px;
+  height: 36px;
+  margin-left: 160px;
+  padding: 0 15px;
+  border-radius: 8px;
+  background-color: #efeeef;
+}
+
+.search-container {
+  display: flex;
+  align-items: center;
+  flex-basis: 100%;
+  height: inherit;
```
속성 순서를 권장 순서로 나열해 놓으니 진짜 가독성 훨씬 좋아지네요~

fullkeem reviewed (commented):
수고하셨습니다!
원희님 코드 덕분에 몰랐던 부분들 많이 공부했습니다.
감사합니다~ :)

Commit:
Refactor : 전체적인 코드 리팩토링
- css disabled 속성 설정 & js 클래스 추가 코드 삭제
- 로그인 유효성 검사 조건 변경 (id도 5글자 이상)
- 검색 토글 delete 버튼 로직 수정 & 자잘한 js 코드 수정
- 불필요한 css 속성 삭제
- 정렬, 마진 등 한가지 속성뿐인 클래스들 정리하여 commom.css로 이동
- common.css에 IR 역할 클래스 선언

Commit:
Add : Mission 10 반응형 구현
- 일정 크기 이상 줄어들면 search input 사라짐
- 프로필 토글의 left값을 resizing 될 때마다 변경

Commit:
Modify: 검색 토글 검색 기능 추가
- 검색 기능 구현
- json 데이터 수정
- 유저 프로필 img파일 수정
- html css link에 type 추가
- IR 추가 및 주석 처리

Commit:
Modify : feed 동적 생성
- 전체 feed들을 데이터를 받아와서 동적 생성
- search input 위치 조정

Commit:
Add : API Mission 3 Database - update
- 인스타 피드 API 생성
- justgram_backend 데이터베이스 생성
- 필요한 테이블 생성
- 필요한 데이터 입력

Commit:
Add : API Mission 3 Database - update
- 인스타 피드 API 생성
- justgram_backend 데이터베이스 생성
- 필요한 테이블 생성
- 필요한 데이터 입력

Commit:
Refactor : 디렉토리 정리
- UI, API, backend로 디렉토리 정리

Commit:
Add : prisma 추가
- prisma package 설치
- prisma와 DB 연결
- prisma와 서버 연결
- 디렉토리 정리

Commit:
✨ Add : 서버에서 받아온 데이터로 피드 동적 구현
- /feeds에 GET메소드로 요청하면 데이터를 받아와서 피드 동적 구현
- prisma is not defined 에러 -> 파일마다 prisma 선언

Commit:
♻️Refactor : 로그인 버튼 활성화 코드 리팩토링
